net/discovery.go

```diff
@@ -83,50 +96,65 @@ func (d *discovery) stop() error {
 	return d.dht.Close()
 }
 
-func (d *discovery) advertise() {
+func (d *discovery) advertiseLoop() {
 	ttl := initialAdvertisementTimeout
 
-	doAdvertise := func() {
-		log.Debug("advertising in the DHT...")
-		err := d.dht.Bootstrap(d.ctx)
-		if err != nil {
-			log.Warnf("failed to bootstrap DHT: err=%s", err)
-			return
-		}
-
-		for _, provides := range d.provides {
-			ttl, err = d.rd.Advertise(d.ctx, string(provides))
-			if err != nil {
-				log.Debugf("failed to advertise in the DHT: err=%s", err)
-				ttl = tryAdvertiseTimeout
-				return
-			}
-		}
-
-		ttl, err = d.rd.Advertise(d.ctx, "")
-		if err != nil {
-			log.Debugf("failed to advertise in the DHT: err=%s", err)
-			ttl = tryAdvertiseTimeout
-			return
-		}
-
-		ttl = defaultAdvertiseTTL
-	}
+	ttl = d.advertise(ttl)
```

Comment:
Removing line 100 and passing the constant here might be slightly more readable.

Reply:
What does `ttl` stand for? It usually means "time to live" and is used for a countdown until something is discarded. This seems more like a `nextRefresh` duration.

Reply:
Ohhh... I'm seeing where the name came from. They are abusing that name too:
https://github.com/libp2p/go-libp2p/blob/v0.21.0/p2p/discovery/routing/routing.go#L32-L37
It was probably correctly used on the DHT record, and then some programmer thought it was a good idea to use it on a value that has almost nothing in common with the time of the DHT record's expiration.